enterprise/server/remote_execution/containers/docker/docker.go

```diff
@@ -401,6 +405,15 @@ func (r *dockerCommandContainer) Exec(ctx context.Context, command *repb.Command
 		res = r.exec(ctx, command, stdin, stdout)
 		return res.Error
 	})
+	// Translate killedExitCode and populate the error field to match the
+	// commandutil.Run semantics.
+	// TODO(bduffany): Find a way to detect whether the exec process was killed
```

Comment:
Could we set a variable in `Remove` to indicate that we are killing the container and then check that value here to limit this to only the shutdown scenario?